Code/Framework/AzCore/AzCore/Serialization/SerializeContext.cpp

```diff
@@ -888,2398 +878,2612 @@ namespace AZ
-                        // if the member is a pointer type, then the pointer could be a derived type,
-                        // otherwise we need the uuids to be exactly the same.
-                        if (childElement->m_flags & SerializeContext::ClassElement::FLG_POINTER)
+                        const AZ::SerializeContext::ClassData* dynamicTypeMetadata = FindClassData(dynamicFieldDesc->m_typeId);
+                        if (dynamicTypeMetadata)
                         {
-                            // Verify that a downcast is possible
-                            bool downcastPossible = m_element.m_id == childElement->m_typeId;
-                            downcastPossible = downcastPossible || (m_classData->m_azRtti && childElement->m_azRtti && m_classData->m_azRtti->IsTypeOf(childElement->m_azRtti->GetTypeId()));
-                            if (downcastPossible)
-                            {
-                                classElement = *childElement;
-                                elementFound = true;
-                            }
+                            AZ::SerializeContext::ClassElement dynamicElementData;
+                            dynamicElementData.m_name = "m_data";
+                            dynamicElementData.m_nameCrc = AZ_CRC("m_data", 0x335cc942);
+                            dynamicElementData.m_typeId = dynamicFieldDesc->m_typeId;
+                            dynamicElementData.m_dataSize = sizeof(void*);
+                            dynamicElementData.m_offset = reinterpret_cast<size_t>(&(reinterpret_cast<AZ::DynamicSerializableField const volatile*>(0)->m_data));
+                            dynamicElementData.m_azRtti = nullptr; // we won't need this because we always serialize top classes.
+                            dynamicElementData.m_genericClassInfo = FindGenericClassInfo(dynamicFieldDesc->m_typeId);
+                            dynamicElementData.m_editData = nullptr; // we cannot have element edit data for dynamic fields.
+                            dynamicElementData.m_flags = ClassElement::FLG_DYNAMIC_FIELD | ClassElement::FLG_POINTER;
+                            EnumerateInstance(callContext, &dynamicFieldDesc->m_data, dynamicTypeMetadata->m_typeId, dynamicTypeMetadata, &dynamicElementData);
                         }
                         else
                         {
-                            if (m_element.m_id == childElement->m_typeId)
-                            {
-                                classElement = *childElement;
-                                elementFound = true;
-                            }
+                            AZ_Error("Serialization", false, "Failed to find class data for 'Dynamic Serializable Field' with type=%s address=%p. Make sure this type is reflected, \
+                                otherwise you will lose data during serialization!\n", dynamicFieldDesc->m_typeId.ToString<AZStd::string>().c_str(), dynamicFieldDesc->m_data);
```

Comment:
I think this will include the entire string so a huge gap in between `reflected,` and `otherwise` will appear in the message.
```suggestion
                            AZ_Error("Serialization", false, "Failed to find class data for 'Dynamic Serializable Field' with type=%s address=%p. Make sure this type is reflected, "\
                                "otherwise you will lose data during serialization!\n", dynamicFieldDesc->m_typeId.ToString<AZStd::string>().c_str(), dynamicFieldDesc->m_data);
```

Reply:
In that case the backslash isn't needed then.